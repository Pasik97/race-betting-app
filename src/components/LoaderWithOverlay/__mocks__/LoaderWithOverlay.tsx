const mock = () => '(LoaderWithOverlay /)';

export default mock;
