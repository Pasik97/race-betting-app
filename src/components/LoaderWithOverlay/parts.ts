import styled, { keyframes } from "styled-components";

export const Overlay = styled.div`
   left: 0;
   top: 0;
   width: 100%;
   height: 100%;
   position: absolute;
   background: #eeeeee;
   opacity: 0.44;
   z-index: 10;
`

export const OverlayInner = styled.div`
   left: 0;
   top: 0;
   width: 100%;
   height: 100%;
   position: absolute;
`;

export const OverlayContent = styled.div`
   left: 50%;
   position: absolute;
   top: 50%;
   transform: translate(-50%, -50%);
`;

const spin = keyframes`
   100% {
      transform: rotate(360deg);
   }
`;

export const Spinner = styled.span`
   width: 150px;
   height: 150px;
   display: inline-block;
   border-width: 5px;
   border-color: rgba(255, 255, 255, 0.05);
   border-top-color: #fff;
   animation: ${spin} 1s infinite linear;
   border-radius: 100%;
   border-style: solid;
`;
