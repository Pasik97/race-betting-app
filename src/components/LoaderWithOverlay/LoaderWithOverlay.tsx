import React from 'react';
import * as P from './parts';

interface LoaderWithOverlayProps {
   loading: boolean;
}

const LoaderWithOverlay: React.FC<LoaderWithOverlayProps> = ({ loading }) => !loading ? null : (
   <P.Overlay>
      <P.OverlayInner>
         <P.OverlayContent>
            <P.Spinner />
         </P.OverlayContent>
      </P.OverlayInner>
   </P.Overlay>
);

export default LoaderWithOverlay;
