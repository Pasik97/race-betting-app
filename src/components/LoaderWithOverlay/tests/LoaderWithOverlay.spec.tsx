import { create } from "react-test-renderer";
import LoaderWithOverlay from "../LoaderWithOverlay";

describe('<LoaderWithOverlay />', () => {
   it('should render divs if loading prop is set to true', () => {
      const component = create(<LoaderWithOverlay loading />);

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should null if loading prop is set to false', () => {
      const component = create(<LoaderWithOverlay loading={false} />);

      expect(component.toJSON()).toBeNull();
   });
});
