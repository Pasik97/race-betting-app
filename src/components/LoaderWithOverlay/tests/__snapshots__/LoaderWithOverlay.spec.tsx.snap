// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderWithOverlay /> should render divs if loading prop is set to true 1`] = `
<div
  className="sc-bdvvtL kGpeYK"
>
  <div
    className="sc-gsDKAQ cFIvzm"
  >
    <div
      className="sc-dkPtRN gYxWmj"
    >
      <span
        className="sc-hKwDye bNnLqN"
      />
    </div>
  </div>
</div>
`;
